### PR TITLE
Added `ContextFeatures.SpecialPostfixOperators`. (fixes #1475)

### LINKED ISSUE #1475: Add version check for PR #1459
Reminder @SlavaRa 

Quite probably having some ContextFeatures property that is enabled if Haxe >= 3.3.

### PR DESCRIPTION
**Pull request by SlavaRa**

### CODE REVIEW

File: External/Plugins/HaXeContext/Context.cs [hunk 1]
```diff
@@ -241,13 +241,15 @@ public void SetHaxeEnvironment(string sdkPath)
...
+            path = path.TrimStart(';');
             path = sdkPath + ";" + path;
-            if (neko != null) path = neko.TrimEnd(new char[] { '/', '\\' }) + ";" + path;
+            if (neko != null) path = neko.TrimEnd('/', '\\') + ";" + path;
             Environment.SetEnvironmentVariable("PATH", path);
             currentEnv = sdkPath;
 
             LoadMetadata();
+            if (GetCurrentSDKVersion().IsGreaterThanOrEquals(new SemVer("3.3.0"))) features.SpecialPostfixOperators = new[] {'!'};
+            else features.SpecialPostfixOperators = new char[0];
```
Comment (Neverbirth) [inline External/Plugins/HaXeContext/Context.cs]: It seems the else is not needed since it's already initialized with an empty array in ContextFeatures itself, isn't it?

File: External/Plugins/ASCompletion/Completion/ASComplete.cs [hunk 2]
```diff
@@ -3496,22 +3497,18 @@ private static ASExpr GetExpression(ScintillaControl sci, int position, bool ign
...
-                                expression.Separator = ' ';
-                                break;
-                            }
-                            genCount--;
+                            position--;
+                            expression.Separator = ' ';
+                            break;
                         }
-                        else sb.Insert(0, c);
+                        genCount--;
```
Comment (Neverbirth) [inline External/Plugins/ASCompletion/Completion/ASComplete.cs]: Why this changes in this `else if` condition? Seems out of the scope of this PR. What does it improve?

File: External/Plugins/HaXeContext/Context.cs (hunk 1 above)
Comment (SlavaRa) [inline External/Plugins/HaXeContext/Context.cs]: Due to the possibility of switching to a version of the SDK version below 3.3.0

File: External/Plugins/ASCompletion/Completion/ASComplete.cs (hunk 2 above)
Comment (SlavaRa) [inline External/Plugins/ASCompletion/Completion/ASComplete.cs]: Sorry, Apparently accidentally got into this commit, but in any case, in the construction ```else if (hasGenerics && (genCount> 0 || c == '<'))``` genCount can not be greater than zero

File: External/Plugins/HaXeContext/Context.cs (hunk 1 above)
Comment (Neverbirth) [inline External/Plugins/HaXeContext/Context.cs]: Ah, I see.
